exercises/book-store/src/test/java/BookStoreTest.java

```diff
@@ -0,0 +1,105 @@
+import static org.junit.Assert.assertEquals;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import org.junit.Test;
+
+public class BookStoreTest{
+
+	@Test
+	public void Basket_with_single_book(){
+		Integer[] p = {1};
+		ArrayList<Integer> books = new ArrayList<>(Arrays.asList(p));
+		assertEquals(8,BookStore.CalculateTotalCost(books),2);
+	}
+
+
+	@Test
+	public void Basket_with_two_of_same_book(){
+		Integer[] p = {1,1};
+		ArrayList<Integer> books = new ArrayList<>(Arrays.asList(p));
+		assertEquals(16,BookStore.CalculateTotalCost(books),2);
+	}
+
+
+	@Test
+	public void Empty_basket(){
+		ArrayList<Integer> books = new ArrayList<>();
+		assertEquals(0,BookStore.CalculateTotalCost(books),2);
+	}
+
+
+	@Test
+	public void Basket_with_two_different_books(){
+		Integer[] p = {1,2};
+		ArrayList<Integer> books = new ArrayList<>(Arrays.asList(p));
+
+		assertEquals(15.20,BookStore.CalculateTotalCost(books),2);
+	}
+
+
+	@Test
+	public void Basket_with_three_different_books(){
+		Integer[] p = {1,2,3};
+		ArrayList<Integer> books = new ArrayList<>(Arrays.asList(p));
+		assertEquals(21.6,BookStore.CalculateTotalCost(books),2);
+	}
+
+
+	@Test
+	public void Basket_with_four_different_books(){
+		Integer[] p = {1,2,3,4};
+		ArrayList<Integer> books = new ArrayList<>(Arrays.asList(p));
+		assertEquals(25.6,BookStore.CalculateTotalCost(books),2);
+
+	}
+
+
+	@Test
+	public void Basket_with_five_different_books(){
+		Integer[] p = {1,2,3,4,5};
+		ArrayList<Integer> books = new ArrayList<>(Arrays.asList(p));
+		assertEquals(30,BookStore.CalculateTotalCost(books),2);
+	}
+
+
```

Comment:
Looks like from this point on, the test suite diverges from the canonical test suite for this exercise.  Is that intentional?

All the remaining tests are extra, and these tests from [the canonical spec](https://github.com/exercism/x-common/blob/master/exercises/book-store/canonical-data.json) are missing:
```json
         {
           "description": "Two groups of four is cheaper than group of five plus group of three",
           "basket": [1,1,2,2,3,3,4,5],
           "targetgrouping": [[1,2,3,4],[1,2,3,5]],
           "expected": 51.20
         },
         {
           "description": "Group of four plus group of two is cheaper than two groups of three",
           "basket": [1,1,2,2,3,4],
           "targetgrouping": [[1,2,3,4],[1,2]],
           "expected": 40.8
         },
         {
           "description": "Two each of first 4 books and 1 copy each of rest",
           "basket": [1,1,2,2,3,3,4,4,5],
           "targetgrouping": [[1,2,3,4,5],[1,2,3,4]],
           "expected": 55.60
         },
         {
           "description": "Two copies of each book",
           "basket": [1,1,2,2,3,3,4,4,5,5],
           "targetgrouping": [[1,2,3,4,5],[1,2,3,4,5]],
           "expected": 60.00
         },
         {
           "description": "Three copies of first book and 2 each of remaining",
           "basket": [1,1,2,2,3,3,4,4,5,5,1],
           "targetgrouping": [[1,2,3,4,5],[1,2,3,4,5],[1]],
           "expected": 68.00
         },
         {
           "description": "Three each of first 2 books and 2 each of remaining books",
           "basket": [1,1,2,2,3,3,4,4,5,5,1,2],
           "targetgrouping": [[1,2,3,4,5],[1,2,3,4,5],[1,2]],
           "expected": 75.20
         }
```